src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -2,44 +2,48 @@ package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
   val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
 
   fun superficie() = ancho * largo
   fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
+    if (ancho > largo) this.superficie() / 5 else this.superficie() / 3 + largo
 
   fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
-    } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
+    if (plantas.size == this.cantidadMaximaPlantas()) {
+      throw Exception("Ya no hay lugar en esta parcela")
+    } else if (horasSolPorDia >= planta.horasDeSolQueTolera() + 2) {
+      throw Exception("No se puede plantar esto acá, se va a quemar")
     } else {
       plantas.add(planta)
-      cantidadPlantas += 1
     }
   }
+  fun parcelaSemillera() =
+    plantas.all {
+      planta -> planta.daSemillas()
+    }
+
+  fun parcelaTieneComplicaciones(parcela: Parcela) =
+          plantas.any { it.horasDeSolQueTolera() < horasSolPorDia }
+
 }
 
 class Agricultora(val parcelas: MutableList<Parcela>) {
-  var ahorrosEnPesos = 20000
 
-  // Suponemos que una parcela vale 5000 pesos
-  fun comprarParcela(parcela: Parcela) {
-    if (ahorrosEnPesos >= 5000) {
+  fun agregar(parcela: Parcela) {
       parcelas.add(parcela)
-      ahorrosEnPesos -= 5000
-    }
   }
 
   fun parcelasSemilleras() =
     parcelas.filter {
-      parcela -> parcela.plantas.all {
-        planta -> planta.daSemillas()
-      }
+      parcela -> parcela.parcelaSemillera()
     }
 
+  fun parcelaConMasLugar() =
+    parcelas.maxBy { it.cantidadMaximaPlantas() - it.plantas.size }!!
+
   fun plantarEstrategicamente(planta: Planta) {
-    val laElegida = parcelas.maxBy { it.cantidadMaximaPlantas() - it.cantidadPlantas }!!
-    laElegida.plantas.add(planta)
+    val laElegida = parcelaConMasLugar()
+
+    laElegida.plantar(planta)
```

Comment:
No es necesario usar un `val` acá, podés directamente encadenar los métodos:
```suggestion
    parcelaConMasLugar().plantar(planta)
```